network-client-okhttp/src/main/java/io/ably/lib/network/OkHttpWebSocketEngineFactory.java

```diff
@@ -0,0 +1,13 @@
+package io.ably.lib.network;
+
+public class OkHttpWebSocketEngineFactory implements WebSocketEngineFactory {
+    @Override
```

Comment:
Can we rename `WebsocketEngineFactory` to `WebsocketClientFactory`. This should also be reflected across all interface impl. `WebsocketEngineFactory` term feels confusing.